[docs] Clarify purpose of reproducibility docs (#2756)
huggingface/diffusers: docs/source/en/using-diffusers/reproducibility.mdx

print(np.abs(image).sum())
```

Running the code above prints one value, but if you run it again you get a different value. What is going on here? 

Every time the pipeline is run, [`torch.randn`](https://pytorch.org/docs/stable/generated/torch.randn.html) uses a different random seed to create Gaussian noise which is denoised stepwise. This leads to a different result each time it is run, which is great for diffusion pipelines since it generates a different random image each time.

> ### CPU

Reviewer: The transition to talking about `CPU` is not yet clear at this point, as we've only set the stage for the problem. I'd maybe suggest something like this:

```suggestion

But what if you really need to generate reproducible results for your project? The solution to this problem is nuanced and depends on the hardware you use.

### CPU
```

> But if you want to reproduce the same image on a CPU, you'll need to use a PyTorch [`Generator`](https://pytorch.org/docs/stable/generated/torch.randn.html) and set a seed:

Reviewer: ```suggestion
To generate reproducible results on a CPU, you'll need to use a PyTorch [`Generator`](https://pytorch.org/docs/stable/generated/torch.randn.html) and set a seed:
```
